From: Jeremy Bicha <jbicha@debian.org>
Date: Thu, 10 Mar 2022 07:51:03 -0500
Subject: Revert "status/bluetooth: Use JS Set for tracking signal
 connections"

This reverts commit 34dcf2f7b13553fcf2e49ac42f95a7f0aec03a2b.

Ubuntu 22.04 LTS isn't using the new gnome-bluetooth series
---
 js/ui/status/bluetooth.js | 25 +++++++++----------------
 1 file changed, 9 insertions(+), 16 deletions(-)

diff --git a/js/ui/status/bluetooth.js b/js/ui/status/bluetooth.js
index 187e106..05050d9 100644
--- a/js/ui/status/bluetooth.js
+++ b/js/ui/status/bluetooth.js
@@ -59,14 +59,10 @@ class Indicator extends PanelMenu.SystemIndicator {
         this._syncId = 0;
         this._adapter = null;
 
-        this._deviceNotifyConnected = new Set();
-
-        const deviceStore = this._client.get_devices();
-        for (let i = 0; i < deviceStore.get_n_items(); i++)
-            this._connectDeviceNotify(deviceStore.get_item(i));
-
+        this._store = this._client.get_devices();
+        this._deviceNotifyConnected = [];
         this._client.connect('device-removed', (c, path) => {
-            this._deviceNotifyConnected.delete(path);
+            this._deviceNotifyConnected[path] = false;
             this._queueSync.bind(this);
         });
         this._client.connect('device-added', (c, device) => {
@@ -87,25 +83,21 @@ class Indicator extends PanelMenu.SystemIndicator {
     }
 
     _connectDeviceNotify(device) {
-        const path = device.get_object_path();
-
-        if (this._deviceNotifyConnected.has(path))
+        if (this._deviceNotifyConnected[device.get_object_path()] !== true)
             return;
-
         device.connect('notify::alias', this._queueSync.bind(this));
         device.connect('notify::paired', this._queueSync.bind(this));
         device.connect('notify::trusted', this._queueSync.bind(this));
         device.connect('notify::connected', this._queueSync.bind(this));
-
-        this._deviceNotifyConnected.add(path);
+        this._deviceNotifyConnected.push([device.get_object_path(), true]);
     }
 
     _getDeviceInfos() {
-        const deviceStore = this._client.get_devices();
         let deviceInfos = [];
+        const numDevices = this._store.get_n_items();
 
-        for (let i = 0; i < deviceStore.get_n_items(); i++) {
-            const device = deviceStore.get_item(i);
+        for (let i = 0; i < numDevices; i++) {
+            const device = this._store.get_item(i);
 
             if (device.paired || device.trusted) {
                 deviceInfos.push({
@@ -113,6 +105,7 @@ class Indicator extends PanelMenu.SystemIndicator {
                     name: device.alias,
                 });
             }
+            this._connectDeviceNotify(device);
         }
 
         return deviceInfos;
